packages/core/content-type-builder/admin/src/components/SelectComponent.tsx

```diff
@@ -66,8 +72,11 @@ export const SelectComponent = ({
   );
 
   if (isAddingAComponentToAnotherComponent) {
-    options = options.filter((option) => {
-      return !componentsThatHaveOtherComponentInTheirAttributes.includes(option.uid);
+    options = options.filter(({ uid }: any) => {
+      const maxDepth = getChildrenMaxDepth(uid, componentsThatHaveOtherComponentInTheirAttributes);
+      const componentDepth = getComponentDepth(targetUid, nestedComponents);
+      const totalDepth = maxDepth + componentDepth;
+      return totalDepth <= MAX_COMPONENT_DEPTH;
```

Comment:
I'm just wondering if we should provide feedback to the user about why
components have been excluded from the list?

I guess we had the same lack of feedback problem before but it was probably more
obvious what was going on as any nesting was filtered out (is that right?)

Now the max depth is set at 6 it could be harder to work out why the filtering
is happening. wdyt?

Reply:
yeah some form of feedback would've been nice,
however it's unclear to me how we could give feedback either, the current way things look
there isn't much space or place to display useful, info, I mean maybe I could put a simple general message such as `components that would exceed the nested limit of 6 are excluded from the list`

Reply:
Yeah it's a bit tricky maybe we should run this by Marco and track it as an improvement after this is merged ?